File: Code_C#.NET/MP06_ChuyenRGBsangCMYK.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Emgu.CV;
using Emgu.CV.Structure;
using Emgu.Util;

namespace MP02_TachAnhMauRGB_EmguCV
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            //Tao bien chua duong dan luu hinh cua minh
            Image<Bgr, byte> hinhgoc = new Image<Bgr, byte>(@"H:\Code\Photo\girl.jpg");

            //Hien thi hinh goc trong picBox_Hinhgoc da tao
            imageBox_Hinhgoc.Image = hinhgoc;
            Image<Bgr, byte> cyan = new Image<Bgr, byte>(hinhgoc.Width, hinhgoc.Height);
            Image<Bgr, byte> magenta = new Image<Bgr, byte>(hinhgoc.Width, hinhgoc.Height);
            Image<Bgr, byte> yellow = new Image<Bgr, byte>(hinhgoc.Width, hinhgoc.Height);
            Image<Bgr, byte> black = new Image<Bgr, byte>(hinhgoc.Width, hinhgoc.Height);

            //Moi hinh la mot ma tran 2 chieu nen se dung 2 vong lap for
            //de doc het cac diem anh co trong hinh
            for (int x = 0; x < hinhgoc.Width; x++)
            {
                for (int y = 0; y < hinhgoc.Height; y++)
                {
                    //doc gia tri pixel tai diem anh co vi tri (x,y)
                    Bgr pixelValue = hinhgoc[y, x];

                    //Tao ra 3 bien R,G,B de lay gia tri RED, GREEN, BLUE tai tuong pixel
                    //Tao bien K de gan gia tri nho nhat cua R,G,B vao
                    byte R = (byte)pixelValue.Red;
                    byte G = (byte)pixelValue.Green;
                    byte B = (byte)pixelValue.Blue;
                    byte K = Math.Min(R, Math.Min(G, B));

                    //set gia tri pixel doc duoc cho cac hinh chua
                    //cac kenh mau tuong ung
                    //Cyan la ket hop cua G va B
                    cyan.Data[y, x, 2] = 0;
                    cyan.Data[y, x, 1] = (Byte)G;
                    cyan.Data[y, x, 0] = (Byte)B;

                    //Magenta la ket hop cua R va B
                    magenta.Data[y, x, 2] = (Byte)R;
                    magenta.Data[y, x, 1] = 0;
                    magenta.Data[y, x, 0] = (Byte)B;

                    //Yellow la ket hop cua R va G
                    yellow.Data[y, x, 2] = (Byte)R;
                    yellow.Data[y, x, 1] = (Byte)G;
                    yellow.Data[y, x, 0] = 0;

                    //Black tao ra khi gan gia tri min(R,G,B) vao ca 3 kenh R,G,B
                    black.Data[y, x, 2] = (Byte)K;
                    black.Data[y, x, 1] = (Byte)K;
                    black.Data[y, x, 0] = (Byte)K;



                }
            }
            //Hien thi cac kenh mau len imageBox
            imageBox_C.Image = cyan;
            imageBox_M.Image = magenta;
            imageBox_Y.Image = yellow;
            imageBox_K.Image = black;
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void imageBox_Hinhgoc_Click(object sender, EventArgs e)
        {

        }
    }
}

File: Codepython/MP06_ChuyenRGBsangCMYK.py
import cv2 #thu vien xu li anh
import numpy as np  #thu vien toan hoc

#Đọc ảnh màu dùng thư viện OpenCV
img = cv2.imread('girl.jpg', cv2.IMREAD_COLOR)

#Lay kich thuoc cua anh
height, width, channel = img.shape

#Khai bao 3 bien de chua gia tri 4 kenh C,M,Y,K
cyan = np.zeros(( height, width,3), np.uint8)
magenta = np.zeros(( height, width, 3), np.uint8)
yellow = np.zeros(( height, width, 3), np.uint8)
black = np.zeros(( height, width, 3), np.uint8)

#Ban dau set zero cho tat ca cac diem anh co trong 3 kenh
cyan[:] = [0,0,0]
magenta[:] = [0,0,0]
yellow[:] = [0,0,0]
black[:] = [0,0,0]

#Dung vong for de doc het cac diem anh
for x in range(width):
    for y in range(height):

        #Lay gia tri R,G,B cua tung pixel va gan vao cac bien R,G,B
        R = img[y,x,2]
        G = img[y,x,1]
        B = img[y,x,0]
        #Gan gia tri min cua R,G,B vao bien K
        K = min(R,G,B)

        #Cyan la ket hop cua Green va Blue
        cyan[y,x,1] = G
        cyan[y,x,0] = B

        #Magenta la ket hop cua Red va Blue
        magenta[y,x,2] = R
        magenta[y,x,0] = B

        #Yellow la ket hop cua Red va Green
        yellow[y,x,2] = R
        yellow[y,x,1] = G

        #Black tao nen khi gan min(R,G,B) vao ca 3 kenh R,G,B
        black[y,x,2] = K
        black[y,x,1] = K
        black[y,x,0] = K

#Hien thi 4 tam hinh tren cung mot cua so
vert=np.concatenate((cyan,magenta),axis=0)
vert1=np.concatenate((yellow,black),axis=0)
vert2=np.concatenate((vert,vert1),axis=1)

#Dieu chinh kich thuoc cua so hien thi
CMYK_program = cv2.resize(vert2,(700,700))

#Hiển thị hình dùng thư viện OpenCV
cv2.imshow('Anh goc', img)
cv2.imshow('Chuyen RGB sang CMYK', CMYK_program)

#Bấm phím bất kì để đóng cửa sổ hiển thị hình
cv2.waitKey(0)

#Giai phong bo nho
cv2.destroyAllWindows()